pkg/imagestream/imagestream.go

```diff
@@ -160,6 +160,46 @@ func (builder *Builder) GetDockerImage(imageTag string) (string, error) {
 		imageTag, builder.Definition.Name, builder.Definition.Namespace)
 }
 
+// HasTag checks if the imageStream has a specific tag.
+func (builder *Builder) HasTag(tagName string) (bool, error) {
```

Comment:
I think that the HasTag checks can be done with GetDockerImage . If the tag does not exist , it will give same message. 

I  would try to use that function, because it covers the same functionality

Reply:
In fact, the `GetDockerImage` checks only the .spec. In our case we need to check the .status for the ImageStream. 

In this case just rename the function for `HasTagInStatus` so that it is clear that the check is in different branch